pkg/resources/common/common.go

```diff
@@ -94,7 +84,7 @@ func (c *cachedVerbChecker) IsRulesAllowed(rules []v1.PolicyRule, resolver valid
 	return err
 }
 
-// HasVerb returns if the request have the overrideVerb. Defaults to false if it hasn't been checked.
-func (c *cachedVerbChecker) HasVerb() bool {
+// HasVerb returns if the request has the overrideVerb. Defaults to false if it hasn't been checked.
```

Comment:
Right now this won't check if the user has an override verb if IsRulesAllowed hasn't been called. You do note this  in the function doc string, so you can keep it this way if you want, but I think the better UX would be to check if the user has the verb if we haven't already, given that the object should have all of the tools necessary to do so.